### PR TITLE
Fix optional model in tabular tasks

### PR DESCRIPTION
**Pull request by Wauplin**
Fix https://github.com/huggingface/huggingface_hub/issues/2015. Thanks @abidlabs for noticing :pray:

### CODE REVIEW

File: src/huggingface_hub/inference/_client.py [hunk 1]
```diff
@@ -1063,16 +1063,17 @@ def table_question_answering(
...
 
         Args:
             table (`Dict[str, Any]`):
                 Set of attributes to classify.
-            model (`str`):
-                The model to use for the tabular-classification task. Can be a model ID hosted on the Hugging Face Hub or a URL to
-                a deployed Inference Endpoint.
+            model (`str`, *optional*):
+                The model to use for the tabular classification task. Can be a model ID hosted on the Hugging Face Hub or a URL to
+                a deployed Inference Endpoint. If not provided, the default recommended text classification model will be used.
```
Comment (abidlabs) [inline src/huggingface_hub/inference/_client.py:1075]:
```suggestion
                a deployed Inference Endpoint. If not provided, the default recommended tabular classification model will be used.
```

Comment (abidlabs) [reply src/huggingface_hub/inference/_client.py:1075]: Note that this docstring is a bit different than the one for e.g. summarization

```
                The model to use for inference. Can be a model ID hosted on the Hugging Face Hub or a URL to a deployed
                Inference Endpoint. This parameter overrides the model defined at the instance level. Defaults to None.
```

File: src/huggingface_hub/inference/_client.py [hunk 2]
```diff
@@ -1107,16 +1108,17 @@ def tabular_classification(self, table: Dict[str, Any], *, model: str) -> List[s
...
 
         Args:
             table (`Dict[str, Any]`):
                 Set of attributes stored in a table. The attributes used to predict the target can be both numerical and categorical.
-            model (`str`):
-                The model to use for the tabular-regression task. Can be a model ID hosted on the Hugging Face Hub or a URL to
-                a deployed Inference Endpoint.
+            model (`str`, *optional*):
+                The model to use for the tabular regression task. Can be a model ID hosted on the Hugging Face Hub or a URL to
+                a deployed Inference Endpoint. If not provided, the default recommended text classification model will be used.
```
Comment (abidlabs) [inline src/huggingface_hub/inference/_client.py:1120]:
```suggestion
                a deployed Inference Endpoint. If not provided, the default recommended tabular regression model will be used.
```

**Comment by abidlabs** [comment]
LGTM thanks for the quick fix @Wauplin! Just a couple of nits regarding the docstrings above

**Comment by Wauplin** [comment]
Thanks for the careful review @abidlabs!

**Pull request merged**